### PR TITLE
adding ReactiveFindByIndexNameSessionRepository

### PR DESCRIPTION
**Pull request by ayudovin**
A reactive equivalent to FindByIndexNameSessionRepository. This [enhancement](https://github.com/spring-projects/spring-session/issues/914)

### CODE REVIEW

**Comment by ayudovin** [comment]
@vpavic, Do i have chance thtat It'll be merged ?

**Comment by vpavic** [comment]
Thanks for the PR @ayudovin - we'll give this a closer look when we start working on 2.2.

File: spring-session-data-redis/src/main/java/org/springframework/session/data/redis/ReactiveRedisOperationsSessionRepository.java [hunk 1]
```diff
@@ -192,6 +198,24 @@ private String getSessionKey(String sessionId) {
...
+	}
+
+	@Override
+	public Flux<Tuple2<String, RedisSession>> findByIndexNameAndIndexValue(String indexName, String indexValue) {
+		if (!PRINCIPAL_NAME_INDEX_NAME.equals(indexName)) {
+			return Flux.empty();
+		}
+		String principalKey = getPrincipalKey(indexValue);
+		return this.sessionRedisOperations.opsForSet()
+				.scan(principalKey)
```
Comment (vpavic) [inline spring-session-data-redis/src/main/java/org/springframework/session/data/redis/ReactiveRedisOperationsSessionRepository.java]: I'm afraid this isn't optimal, as you would be scanning for a match each time, rather having an index that's updated on write and have reads simple.

**Comment by vpavic** [comment]
Closing due to lack of feedback. Please comment back if you can provide more details and we can re-open the issue.

**Pull request closed without merging**